File: src/main/java/com/koddy/server/auth/domain/model/Authenticated.java
package com.koddy.server.auth.domain.model;

import static com.koddy.server.member.domain.model.Role.MENTEE;
import static com.koddy.server.member.domain.model.Role.MENTOR;

public record Authenticated(
        long id,
        String authority
) {
    public boolean isMentor() {
        return MENTOR.getAuthority().equals(authority);
    }

    public boolean isMentee() {
        return MENTEE.getAuthority().equals(authority);
    }
}

File: src/main/java/com/koddy/server/auth/domain/model/Authenticated.kt
package com.koddy.server.auth.domain.model

import com.koddy.server.member.domain.model.Role

data class Authenticated(
    val id: Long,
    val authority: String,
) {
    val isMentor: Boolean
        get() = Role.MENTOR.authority == authority

    val isMentee: Boolean
        get() = Role.MENTEE.authority == authority
}
